File: sort/Insertion Sort.py
# The array is virtually split into a sorted and an unsorted part.
#  Values from the unsorted part are picked and placed at the correct
#  position in the sorted part.


def insertion(arr,n):
    for i in range(1,n):
        for j in range(i):
            if arr[j]>arr[i]:
                temp=arr[j]
                arr[j]=arr[i]
                arr[i]=temp
    return arr

def func():
    arr=list(map(int,input("Enter elements:").split()))
    n=len(arr)
    arr=insertion(arr,n)
    print(arr)

if __name__=='__main__':
    func()

File: sort/Insertion Sort.cpp
#include<iostream>
using namespace std;
int insertion(int arr[],int n)
{
    int temp;
    for(int i=1;i<n;i++)
    {
        for(int j=0;j<i;j++)
        {
            if(arr[j]>arr[i])
            {
                temp=arr[j];
                arr[j]=arr[i];
                arr[i]=temp;
            }
        }
    }
    return *arr;
}
int main()
{
    int n;
    cout<<"Enter number of element in array:";
    cin>>n;
    int arr[n];
    cout<<"Enter array:";
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    *arr=insertion(arr,n);
    for(int i=0;i<n;i++)
    {
        cout<<arr[i]<<' ';
    }
}
